File: General/radius.swift
import Foundation

class Post {
    let id: String              // Unique ID
    let contents: String        // Contents of the post
    let likes: [String]         // Array of User IDs
    let timestamp: UInt         // Timestamp of when the post was made.
}

class User {
    let id: String              // Unique UID
    let feed: [String]          // Array of Post IDs
}

func getLikedPosts(for user: User, posts: [Post]) -> [Post] {
    // TODO
    return []
}

func getFeeds(for users: [User], posts: [Post]) -> [String : Post] {
    // TODO
    return [:]
}

File: General/radius.js
class Post {
    constructor(id, contents, likes, timestamp) {
        this.id = id;                   // Unique ID
        this.contents = contents;       // Contents of the post
        this.likes = likes;             // Array of User IDs
        this.timestamp = timestamp;     // Timestamp of when the post was made.
    }
}

class User {
    constructor(id, feed) {
        this.id = id;                   // Unique UID
        this.feed = feed;               // Array of Post IDs
    }
}

function getLikedPosts(user, posts) {
    // TODO
    return [];
}

function getFeeds(users, posts) {
    // TODO
    return {};
}